plugins/blockvolume/blockprovider/gluster-block/glusterblock.go

```diff
@@ -0,0 +1,314 @@
+	if blockVolOpts.HostVol == "" && !g.HostingVolumeOptions.AutoCreate {
+		err := errors.New("host volume is not provided and auto creation is not enabled")
+		log.WithError(err).Error("failed in creating block volume")
+		return nil, err
+	}
+
+	// If HostingVolume name is not empty, then create block volume with requested size.
+	// If available size is less than requested size then ERROR. Set block related
+	// metadata and volume options if not exists.
+	if blockVolOpts.HostVol != "" {
+		vInfo, err := volume.GetVolume(blockVolOpts.HostVol)
+		if err != nil {
+			log.WithError(err).Error("error in fetching volume info")
+			return nil, err
+		}
+		volInfo = vInfo
+	}
+
+	// If HostingVolume is not specified. List all available volumes and see if any volume is
+	// available with Metadata:block-hosting=yes
+	if blockVolOpts.HostVol == "" {
+		vInfo, err := GetExistingBlockHostingVolume(size)
+		if err != nil {
+			log.WithError(err).Debug("no block hosting volumes present")
+		}
+		volInfo = vInfo
+	}
+
+	// If No volumes are available with Metadata:block-hosting=yes or if no space available to create block
+	// volumes(Metadata:block-hosting-available-size is less than request size), then try to create a new
+	// block hosting Volume with generated name with default size and volume type configured
+	if blockVolOpts.HostVol == "" && volInfo == nil {
+		vInfo, err := CreateBlockHostingVolume(volCreateReq)
+		if err != nil {
+			log.WithError(err).Error("error in auto creating block hosting volume")
+			return nil, err
+		}
+		if err := StartBlockHostingVolume(vInfo.Name); err != nil {
+			log.WithError(err).Error("error in starting auto created block hosting volume")
+			return nil, err
+		}
+		volInfo = vInfo
+	}
+
+	if _, found := volInfo.Metadata["_block-hosting"]; !found {
+		volInfo.Metadata["_block-hosting"] = "yes"
+	}
+
+	blockHosting := volInfo.Metadata["_block-hosting"]
+
+	if strings.ToLower(blockHosting) == "no" {
+		return nil, errors.New("not a block hosting volume")
+	}
+
+	if _, found := volInfo.Metadata["_block-hosting-available-size"]; !found {
+		volInfo.Metadata["_block-hosting-available-size"] = fmt.Sprintf("%d", g.HostingVolumeOptions.Size)
+	}
+
+	availableSizeInBytes, err := strconv.Atoi(volInfo.Metadata["_block-hosting-available-size"])
+
+	if err != nil {
+		return nil, err
+	}
+
+	if int64(availableSizeInBytes) < size {
+		return nil, fmt.Errorf("available size is less than requested size,request size: %d, available size: %d", size, availableSizeInBytes)
+	}
+
+	req := &api.BlockVolumeCreateReq{
+		HaCount:            blockVolOpts.Ha,
+		AuthEnabled:        blockVolOpts.Auth,
+		FullPrealloc:       blockVolOpts.FullPrealloc,
+		Size:               uint64(size),
+		Storage:            blockVolOpts.Storage,
+		RingBufferSizeInMB: blockVolOpts.RingBufferSizeInMB,
+		Hosts:              hosts,
+	}
+
+	resp, err := g.client.CreateBlockVolume(volInfo.Name, name, req)
```

Comment:
We need to make sure we have the BlockHostingVolume is up and running. Because, if it is not present, then this will fail for sure. If it is already taken care by the events above, then all good.

Reply:
Line 149 is checking that Hosting Volume is in Started State